src/interchange/src/avro/envelope_debezium.rs

```diff
@@ -320,7 +326,15 @@ impl<'a> AvroDecode for DebeziumSourceDecoder<'a> {
                 "file" => {
                     let d = AvroStringDecoder::with_buf(self.file_buf);
                     field.decode_field(d)?;
-                    has_file = true;
+                    file_idx = Some(match self.filenames_to_indices.get(self.file_buf) {
+                        Some(idx) => *idx,
+                        None => {
+                            let n_files = self.filenames_to_indices.len();
+                            self.filenames_to_indices
+                                .insert(std::mem::take(self.file_buf), n_files);
+                            n_files
+                        }
+                    });
```

Comment:
`file_buf` is the buffer that the decoder writes filenames into. It exists (rather than having the decoder return a string) purely to avoid extra allocations in this code. (At any rate, it's not anything new -- this is what we were using to get the filename before).